readme: update and split into various parts

nixbitcoin opened the pull request:
This PR updates the README.md and splits some of its components (examples, tests) into the relative directories.

I wanted to keep it minimal and nice on the eyes.

Closes https://github.com/fort-nix/nix-bitcoin/issues/269

danielabrozzoni commented:
Ack 48b7631df4037a176d373578f756c7986c2f43d5

erikarvstedt commented:
[Fixups](https://github.com/erikarvstedt/nix-bitcoin/commits/update-readme-ea).

With my graphics design background I couldn't resist dabbling with the logo.
It's now a bit more polished and space-efficient. Feel free to ignore it if you don't like it.
Comparison:
![](https://i.imgur.com/nw3kOCs.png)

jonasnick commented on README.md:
```diff
@@ -1,144 +1,97 @@
+* adds non-root user "operator" who has access to client tools (ex. bitcoin-cli, lightning-cli)
+
+In `configuration.nix` the user can enable
+* announcing a clightning onion service and [clightning plugins](https://github.com/lightningd/plugins):
```
How about linking to the plugin section in usage.md here? Would save some space. Perhaps we can keep some highlights here, like clboss.

nixbitcoin replied:
I like seeing all the c-lightning plugins. We can remove this section if we add more plugins and it takes up too much space.

jonasnick commented on README.md:
```diff
@@ -1,144 +1,97 @@
-It should be a reproducible and extensible platform for applications building on Bitcoin.
-
-Examples
+Goals
```
I'd suggest to replace the whole Goals section with something like the following:

Overview
---
A Bitcoin node verifies the Bitcoin protocol and provides ways of interacting with the Bitcoin network. nix-bitcoin nodes are used for a variety of purposes and can serve as personal or merchant wallets, second layer public infrastructure and as backends for Bitcoin applications. In all cases, the aim is to provide security and privacy by default. However, while nix-bitcoin is used in production today, it is still considered experimental.

A full installation of nix-bitcoin is usually deployed either on a dedicated (virtual) machine or runs in a container and is online 24/7. Alternatively, the Nix packages, NixOS modules and configurations can be used independently and combined freely.

nix-bitcoin is built on top of Nix and NixOS because that allows using abstractions powerful enough to keep it highly customizable and maintainable. Testament to this are nix-bitcoin's security features and its potent test framework. However, running nix-bitcoin does not require any previous experience with the nix ecosystem.

jonasnick reviewed (commented):
Largely agree with @erikarvstedt's fixups. Logo looks nice! I wonder if there's some way to make it look better with the dark theme.

nixbitcoin commented:
Very nice logo @erikarvstedt! I like the light theme.

I incorporated most of @erikarvstedt's changes. I also added a sentence in `examples/README.md` that explains what the `configuration.nix` enables by default. (Compromise on https://github.com/erikarvstedt/nix-bitcoin/commit/4645fe8be4f6d195331e1e3defc9ab4487290560#r45518427)

I also replaced the "Goals" section with @jonasnick's "Overview".

erikarvstedt commented:
[Fixups](https://github.com/erikarvstedt/nix-bitcoin/commits/update-readme-ea).

I've removed the white logo background, now it also fits the dark theme:
![](https://i.imgur.com/piEWwo9.png)

nixbitcoin commented:
Squashed

erikarvstedt commented on README.md:
```diff
@@ -1,144 +1,103 @@
-# Run a Python test shell inside a VM node
-./run-tests.sh debug
-print(succeed("systemctl status bitcoind"))
+nix-bitcoin is built on top of Nix and NixOS because that allows using abstractions powerful enough to keep it highly
+customizable and maintainable. Testament to this are nix-bitcoin's robust security features and its potent test
+framework.  However, running nix-bitcoin does not require any previous experience with the Nix ecosystem.
```
Nit, make wording more concise and poignant:
```suggestion
nix-bitcoin is built on top of Nix and NixOS which provide powerful abstractions to keep it highly
customizable and maintainable. Testament to this are nix-bitcoin's robust security features and its potent test
framework.  However, running nix-bitcoin does not require any previous experience with the Nix ecosystem.
```

nixbitcoin replied:
Fixed in bcedf69549652b9d201cc9e02f64c086d6501d14

erikarvstedt reviewed (approved):
ACK bcedf69549652b9d201cc9e02f64c086d6501d14

jonasnick commented:
Good job everyone. Really happy with how this turned out.

erikarvstedt commented:
Agreed. Great work, team!
